Xamarin.Forms.Platform.iOS/Renderers/IndicatorViewRenderer.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System.ComponentModel;
+using UIKit;
+using static Xamarin.Forms.IndicatorView;
+
+namespace Xamarin.Forms.Platform.iOS
+{
+	public class IndicatorViewRenderer : ViewRenderer<IndicatorView, UIPageControl>
+	{
+		UIColor _defaultPagesIndicatorTintColor;
+		UIColor _defaultCurrentPagesIndicatorTintColor;
+
+		protected override void OnElementChanged(ElementChangedEventArgs<IndicatorView> e)
+		{
+			if (e.NewElement != null)
+			{
+				if (Control == null)
+				{
+					SetNativeControl(CreateNativeControl());
+				}
+				UpdatePagesIndicatorTintColor();
+				UpdateCurrentPagesIndicatorTintColor();
+				UpdatePages();
+				UpdateHidesForSinglePage();
+				UpdateCurrentPage();
+			}
+			base.OnElementChanged(e);
+		}
+
+		protected override void OnElementPropertyChanged(object sender, PropertyChangedEventArgs e)
+		{
+			base.OnElementPropertyChanged(sender, e);
+			if (e.PropertyName == IndicatorColorProperty.PropertyName)
+				UpdatePagesIndicatorTintColor();
+			else if (e.PropertyName == SelectedIndicatorColorProperty.PropertyName)
+				UpdateCurrentPagesIndicatorTintColor();
+			else if (e.PropertyName == CountProperty.PropertyName)
+				UpdatePages();
+			else if (e.PropertyName == HidesForSingleIndicatorProperty.PropertyName)
+				UpdateHidesForSinglePage();
+			else if (e.PropertyName == PositionProperty.PropertyName)
+				UpdateCurrentPage();
+		}
+
+		protected override UIPageControl CreateNativeControl()
+			=> new UIPageControl();
+
+		void UpdateCurrentPage()
+			=> Control.CurrentPage = Element.Position;
+
+		void UpdatePages()
+			=> Control.Pages = Element.Count;
+
+		void UpdateHidesForSinglePage()
+			=> Control.HidesForSinglePage = Element.HidesForSingleIndicator;
+
+		void UpdatePagesIndicatorTintColor()
+		{
+			_defaultPagesIndicatorTintColor = _defaultPagesIndicatorTintColor ?? Control.PageIndicatorTintColor;
```

Comment:
Setting `_defaultPagesIndicatorTintColor` and `_defaultCurrentPagesIndicatorTintColor` can be moved inside `OnElementChanged`. Here, the assignment checks will be made every time colors change, but we really need to get the default colors only once.

Reply:
Oh, you are right. By the way, it could lead to a bug because default value is nil if I am not wrong

Reply:
The default value for the class fields is null, but you're using a null coalescing operator, so you should pick Control properties the first time.